server/util.go

```diff
@@ -348,6 +348,9 @@ func dumpTextValue(mysqlType uint8, value types.Datum) ([]byte, error) {
 		return hack.Slice(value.GetMysqlEnum().String()), nil
 	case types.KindMysqlSet:
 		return hack.Slice(value.GetMysqlSet().String()), nil
+	case types.KindMysqlJSON:
+		j := value.GetMysqlJSON()
+		return hack.Slice(j.DumpToString()), nil
```

Comment:
use value.GetMysqlJSON().DumpToString() ?
btw, why not use String() but DumpToString()